ash/src/device.rs

```diff
@@ -2016,21 +2016,15 @@ impl Device {
         &self,
         query_pool: vk::QueryPool,
         first_query: u32,
-        query_count: u32,
         data: &mut [T],
         flags: vk::QueryResultFlags,
     ) -> VkResult<()> {
-        let data_length = query_count as usize;
-        assert!(
-            data_length <= data.len(),
-            "query_count was higher than the length of the slice"
-        );
-        let data_size = mem::size_of::<T>() * data_length;
+        let data_size = mem::size_of_val(data);
```

Comment:
This effectively incorporates #640; not to fix a "bug" (there was no bug as far as I understood, the caller was using the wrong the type and doing it that way would still result in the wrong stride hence the wrong output for `WITH_AVAILABILITY` as result N+1 overwrites N), but simply because this otherwise reads `mem::size_of::<T>() * data.len()` which should be exactly what `mem::size_of_val()` returns.